### ISSUE TITLE
Panic when tearing down iscsi connection

### ISSUE THREAD

**Issue by tobad357** [kubernetes/kubernetes#18744]
Kubelet version: 1.1.3
OS: Centos 7.1

We are currently seeing an issue where each time an iscsi volume gets torn down there is a panic in iscsi_util.go. This prevents us from locking the iscsi target to only allowing one login per volume and can therefor cause (and has caused) data loss due to one volume being accessed by two hosts by mistake

```
W1216 07:18:31.527192    7970 kubelet.go:1588] Orphaned volume "d0ccbd99-a3c4-11e5-8a65-0cc47a4c5dae/data" found, tearing down volume
Dec 16 07:18:31 srv06 kubelet[7970]: E1216 07:18:31.556498    7970 util.go:82] Recovered from panic: "slice bounds out of range" (runtime error: slice bounds out of range)
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/util.go:76
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/util.go:64
Dec 16 07:18:31 srv06 kubelet[7970]: /usr/src/go/src/runtime/asm_amd64.s:402
Dec 16 07:18:31 srv06 kubelet[7970]: /usr/src/go/src/runtime/panic.go:387
Dec 16 07:18:31 srv06 kubelet[7970]: /usr/src/go/src/runtime/panic.go:18
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/iscsi/iscsi_util.go:138
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/iscsi/disk_manager.go:94
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/iscsi/iscsi.go:189
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/iscsi/iscsi.go:185
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1592
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1770
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1987
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:808
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:777
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/util.go:125
Dec 16 07:18:31 srv06 kubelet[7970]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/util.go:126
```

This seems to be the same as #15556.
Are there any plans on backporting that release into the 1.1 branch?

**Comment by tobad357** [comment]
Can confirm this is fixed in 1.2.0-alpha4.

**Comment by jlowdermilk** [comment]
cc @dchen1107

**Comment by dchen1107** [comment]
@saad-ali please triage. Thanks!

**Comment by saad-ali** [comment]
@rootfs Can you please take a look at this.

**Comment by rootfs** [comment]
@saad-ali this needs backport, is it fine to cherry pick https://github.com/kubernetes/kubernetes/pull/15562?

**Comment by saad-ali** [comment]
@rootfs Yes, let's get #15562 cherry picked to the 1.1 release branch. Let me know if you need help with creating the cherry pick.

**Comment by rootfs** [comment]
@saad-ali cool, backport proposed

**Comment by saad-ali** [comment]
This should be fixed with cherry pick of #18744 to `kubernetes:release-1.1` in PR #18875
